src/main.py

```diff
@@ -0,0 +1,119 @@
+# (c) Open Compute Project
+# (c) Google LLC
+# (c) Igor Semenov (goshik92@gmail.com)
+# 
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+
+import ocptv.output as tv
+import argparse
+import sh
+import re
+import os
+import contextlib
+from memtester_parsing import MemtesterObserver
+
+def main():
+    parser = argparse.ArgumentParser(description="OCP-compliant memtester wrapper.")
+    parser.add_argument("--mt_path", default="memtester",
+        help="Path to memtester executable")
+    parser.add_argument("--mt_args", default="",
+        help="Memtester arguments packed into a string. See `man memtester` for more detail")
+    parser.add_argument("--mt_log_filename", default=None,
+        help="Path to the file that will contain raw memtester output (both STDOUT and STDERR)")
+    args = parser.parse_args()
+
+    run = tv.TestRun(name="memtester", version="1.0")
+    dut = tv.Dut(id=sh.hostid().strip(), name=sh.hostname().strip())
+    observer = MemtesterObserver()
+    
+    # Use ExitStack() to avoid deep nesting
+    with contextlib.ExitStack() as estack:
+        estack.enter_context(run.scope(dut=dut))
+        estack.enter_context((step := run.add_step("run-memtester")).scope())
+
+        bad_addrs = step.start_measurement_series(name="bad_addrs")
+        
+        # Check if a supported version of memtester is installed in the system
+        def version_callback(version, is_known):
+            step.add_log(tv.LogSeverity.INFO, "Memtester v{} was found".format(version))
+            if (not is_known):
+                m = "This version of memtester was not tested. Expect parsing errors"
```

Comment:
i think this behavior should be gated by a program arg; to let the user decide if they accept the program possibly failing with errors or refuse to run a nonsupported memtester bin from the start

Reply:
Since the parser is strict about the text's grammar, shouldn't we assume that if a newer memtester version does not cause parsing errors, most likely it will work fine with this wrapper? In case when the parser fails, the user will definitely know that because of the ERROR artifact. The warning log message will give the user some hint about why they got the parsing error. Opinion?

Reply:
UPD: As discussed in the meeting, by default, the execution should fail if the version is not marked as supported. This behavior should be overridible with a flag.

Reply:
Done.